### PR TITLE
Revert "Make Console.ReadKey() distinguish between CR and LF inputs"

### LINKED ISSUE #42418: System.Console.ReadKey() returning Ctrl+J for ENTER on macOS
### Description

See the following code:

```csharp
using System;

var key = Console.ReadKey(true);
Console.WriteLine("KEY: " + key.Key);
Console.WriteLine("MODS: " + key.Modifiers);
```

Run it with `dotnet run` and hit ENTER.

**NOTE: You may need to SPAM hitting enter as I see this `Ctrl`+`J` behavior _only_ when the process is under strain...**

### Configuration

* Which version of .NET is the code running on? **.NET 5 RC1**
* What OS and version, and what distro if applicable? **macOS Catalina**
* What is the architecture (x64, x86, ARM, ARM64)? **x64**
* Do you know whether it is specific to that configuration? **No. It could be others. Not sure.**

### Regression?

Yes this worked before. It worked in .NET 5.0 preview 7. Then a similar bug showed up in .NET 5.0 preview 8 (except it was Ctrl+Enter instead of Ctrl+J)... 

### Other information

This was discovered in [PowerShellEditorServices](https://github.com/PowerShell/PowerShellEditorServices). The end result is the user hit's ENTER in the PowerShell Integrated Console and the console does nothing making the console useless.

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Reverts #37491. Fixes #42418 

cc @tmds @stephentoub @jeffhandley

### CODE REVIEW

**Comment by tmds** [comment]
@eiriktsarpalis I carefully went over the changes, this looks good to me. Did you run the ManualTests and verified they pass?

**Comment by eiriktsarpalis** [comment]
> Did you run the ManualTests and verified they pass?

Yep.

File: src/libraries/System.Console/src/System/ConsolePal.Unix.cs [hunk 1]
```diff
@@ -1210,19 +1219,11 @@ private void AddKey(TermInfo.Database db, string extendedName, ConsoleKey key, b
...
-            {
-                fixed (byte* bufPtr = buffer)
-                {
-                    int result = Interop.CheckIo(Interop.Sys.Read(fd, (byte*)bufPtr + offset, count));
-                    Debug.Assert(result <= count);
-                    return result;
-                }
-            }
-            finally
+            fixed (byte* bufPtr = buffer)
```
Comment (stephentoub) [inline src/libraries/System.Console/src/System/ConsolePal.Unix.cs:1222]: @tmds, after this goes in, you can change back https://github.com/dotnet/runtime/pull/42303/files#diff-142ed2172d119029385e482ff7377130R267, right?

File: src/libraries/System.Console/tests/ManualTests/ManualTests.cs [hunk 2]
```diff
@@ -138,8 +138,6 @@ public static IEnumerable<object[]> GetKeyChords()
             yield return MkConsoleKeyInfo('\x02', ConsoleKey.B, ConsoleModifiers.Control);
             yield return MkConsoleKeyInfo(OperatingSystem.IsWindows() ? '\x00' : '\x02', ConsoleKey.B, ConsoleModifiers.Control | ConsoleModifiers.Alt);
             yield return MkConsoleKeyInfo('\r', ConsoleKey.Enter, (ConsoleModifiers)0);
-            // windows will report '\n' as 'Ctrl+Enter', which is typically not picked up by Unix terminals
-            yield return MkConsoleKeyInfo('\n', OperatingSystem.IsWindows() ? ConsoleKey.Enter : ConsoleKey.J, ConsoleModifiers.Control);
```
Comment (stephentoub) [inline src/libraries/System.Console/tests/ManualTests/ManualTests.cs:142]: Can we keep a "fixed" test here?

**Comment by eiriktsarpalis** [comment]
Per @stephentoub's request I just forced push a change that breaks the PR into two commits: one that contains the actual revert and one that contains necessary fix: accounting for terminals that have the `ICRNL` flag disabled.

File: src/libraries/System.Console/tests/ManualTests/ManualTests.cs (hunk 2 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Console/tests/ManualTests/ManualTests.cs:142]: You mean keep the test for windows only?

Comment (stephentoub) [reply src/libraries/System.Console/tests/ManualTests/ManualTests.cs:142]: I mean continue to validate this behavior on Windows, and have the test validate on Linux/macOS whatever the behavior is there now.

Comment (eiriktsarpalis) [reply src/libraries/System.Console/tests/ManualTests/ManualTests.cs:142]: Done.

**Comment by jeffhandley** [comment]
@eiriktsarpalis and @tmds, before I present https://github.com/dotnet/runtime/pull/42477 for Ask Mode, I'll need you to augment the testing notes there to reflect all the testing you've done.  Those notes could also include the verification that the behavior from #802 that was previously fixed has indeed come back.  Thanks!

**Comment by eiriktsarpalis** [comment]
@jeffhandley done.

**Comment by jeffhandley** [comment]
Thanks, @eiriktsarpalis.  When this PR is signed off, we can run the backport command again and the bot will catch the backport PR up with the needed commits.

**Comment by eiriktsarpalis** [comment]
FYI @adamsitnik

**Comment by jeffhandley** [comment]
@stephentoub and @adamsitnik Could at least one of you review on Wednesday (Sept 23)? Once this is merged, I'll get the RC2 ask mode request sent out.  We need to have #42477 merged in by 10am Thursday (Sept 24) to make RC2.

**Review by stephentoub — APPROVED** [review]

**Review by tmds — APPROVED** [review]

File: src/libraries/System.Console/src/System/ConsolePal.Unix.cs (hunk 1 above)
Comment (tmds) [reply src/libraries/System.Console/src/System/ConsolePal.Unix.cs:1222]: Yes, I've just pushed a commit to that PR to remove the `TermInfo.cs` changes.

**Review by adamsitnik — APPROVED** [review]
LGTM

**Pull request merged**